Spanner/src/Database.php

```diff
@@ -871,11 +873,15 @@ public function runTransaction(callable $operation, array $options = [])
         };
 
         $delayFn = function (\Exception $e) {
-            if (!($e instanceof AbortedException)) {
-                throw $e;
+            if ($e instanceof AbortedException ||
+                (
+                    $e instanceof ServiceException &&
+                    $e->getCode() === Code::INTERNAL &&
+                    strpos($e->getMessage(), 'RST_STREAM') !== false
+                )) {
+                return $e->getRetryDelay();
```

Comment:
nit, but I find this hard to read. How about something like this?
```suggestion
            if ($e instanceof AbortedException) {
                return $e->getRetryDelay();
            }
            if ($e instanceof ServiceException
                && $e->getCode() === Code::INTERNAL
                && strpos($e->getMessage(), 'RST_STREAM') !== false
            ) {
                return $e->getRetryDelay();
            }
```